SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -299,114 +162,118 @@ void USpatialWorkerConnection::ConnectToLocator()
 
 	Worker_ConnectionFuture* ConnectionFuture = Worker_Alpha_Locator_ConnectAsync(WorkerLocator, &ConnectionParams);
 
+	FinishConnecting(ConnectionFuture);
+}
+
+void USpatialWorkerConnection::FinishConnecting(Worker_ConnectionFuture* ConnectionFuture)
+{
 	AsyncTask(ENamedThreads::AnyBackgroundThreadNormalTask, [ConnectionFuture, this]
 	{
-		WorkerConnection = Worker_ConnectionFuture_Get(ConnectionFuture, nullptr);
-
+		Worker_Connection* NewWorkerConnection = Worker_ConnectionFuture_Get(ConnectionFuture, nullptr);
 		Worker_ConnectionFuture_Destroy(ConnectionFuture);
-		if (Worker_Connection_IsConnected(WorkerConnection))
+
+		AsyncTask(ENamedThreads::GameThread, [this, NewWorkerConnection]
 		{
+			this->WorkerConnection = NewWorkerConnection;
+
 			CacheWorkerAttributes();
-			AsyncTask(ENamedThreads::GameThread, [this]
+
+			if (Worker_Connection_IsConnected(NewWorkerConnection))
 			{
-				this->OnConnectionSuccess();
-			});
-		}
-		else
-		{
-			// TODO: Try to reconnect - UNR-576
-			AsyncTask(ENamedThreads::GameThread, [this]
+				OnConnectionSuccess();
+			}
+			else
 			{
-				this->OnConnectionFailure();
-			});
-		}
+				// TODO: Try to reconnect - UNR-576
+				OnConnectionFailure();
+			}
+		});
 	});
 }
 
 SpatialConnectionType USpatialWorkerConnection::GetConnectionType() const
 {
-	// The legacy locator path did not specify PlayerIdentityToken, so if we have one
-	// we can use the new locator workflow.
 	if (!LocatorConfig.PlayerIdentityToken.IsEmpty())
 	{
 		return SpatialConnectionType::Locator;
 	}
-	else if (!LegacyLocatorConfig.LoginToken.IsEmpty())
-	{
-		return SpatialConnectionType::LegacyLocator;
-	}
 	else
 	{
 		return SpatialConnectionType::Receptionist;
 	}
 }
 
-Worker_OpList* USpatialWorkerConnection::GetOpList()
+TArray<Worker_OpList*> USpatialWorkerConnection::GetOpList()
```

Comment:
Alternatively we could have a TryGetOplist that returns a single OpList and we do that until it returns nullptr. Saves allocating an array, but not a big deal.

Reply:
We totally could. That sounds like a good idea.